vital/types/detected_object.cxx

```diff
@@ -38,6 +38,13 @@
 namespace kwiver {
 namespace vital {
 
+detected_object::detected_object( double              confidence,
+                                  detected_object_type_sptr classifications )
+  : m_confidence( confidence )
+  , m_type( classifications )
+  , m_index( 0 )
```

Comment:
This (`m_index`) looks like a good candidate for an inline initializer ([[misc.modern.member_init]](../tree/master/doc/code-style.rst))... Also, consider using brace initializers ([[misc.modern.construct]](../tree/master/doc/code-style.rst))?